src/uu/uniq/src/uniq.rs

```diff
@@ -58,22 +63,26 @@ impl Uniq {
     ) {
         let mut lines: Vec<String> = vec![];
         let mut first_line_printed = false;
-        let delimiters = &self.delimiters;
+        let delimiters = self.delimiters;
         let line_terminator = self.get_line_terminator();
+        let print_delimiters = delimiters != Delimiters::None;
+        let prepend_delimiter = delimiters == Delimiters::Prepend || delimiters == Delimiters::Both;
+        let append_delimiter = delimiters == Delimiters::Append || delimiters == Delimiters::Both;
 
         for line in reader.split(line_terminator).map(get_line_string) {
             if !lines.is_empty() && self.cmp_keys(&lines[0], &line) {
-                let print_delimiter = delimiters == &Delimiters::Prepend
-                    || (delimiters == &Delimiters::Separate && first_line_printed);
+                let print_delimiter = print_delimiters && (prepend_delimiter || first_line_printed);
```

Comment:
Ditto

Reply:
Is the lack of clarity here due to the similarity of the names `print_delimiter` and `print_delimiters`? May be removing `print_delimiters` above and just using `delimiters != Delimiters::None` will make it more readable.

```suggestion
                let print_delimiter =
                    delimiters != Delimiters::None && (prepend_delimiter || first_line_printed);
```
Thinking of way to express this logic with more readable code without comments.